extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/internal/DmxBridgeHandler.java

```diff
@@ -0,0 +1,237 @@
+            case CHANNEL_MUTE:
+                if (command instanceof OnOffType) {
+                    isMuted = ((OnOffType) command).equals(OnOffType.ON);
+                } else {
+                    logger.warn("command {} not supported in channel {}:mute", command.getClass(), this.thing.getUID());
+                }
+                break;
+            default:
+                logger.warn("Channel {} not supported in bridge {}", channelUID.getId(), this.thing.getUID());
+        }
+    }
+
+    /**
+     * get a DMX channel from the bridge
+     *
+     * @param channel
+     *            a BaseChannel that identifies the requested channel
+     * @param thing
+     *            the Thing that requests the channel to track channel usage
+     * @return
+     *         a Channel object
+     */
+    public Channel getDmxChannel(BaseChannel channel, Thing thing) {
+        return universe.registerChannel(channel, thing);
+    }
+
+    /**
+     * remove a thing from all channels in the universe
+     *
+     * @param thing
+     *            the thing that shall be removed
+     */
+    public void unregisterDmxChannels(Thing thing) {
+        universe.unregisterChannels(thing);
+    }
+
+    /**
+     * get the universe associated with this bridge
+     *
+     * @return
+     *         the DMX universe id
+     */
+    public int getUniverseId() {
+        return universe.getUniverseId();
+    }
+
+    /**
+     * rename the universe associated with this bridge
+     *
+     * @param universeId
+     *            the new DMX universe id
+     */
+    protected void renameUniverse(int universeId) {
+        universe.rename(universeId);
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        updateConfiguration();
+    }
+
+    /**
+     * open the connection to send DMX data to
+     */
+    protected abstract void openConnection();
+
+    /**
+     * close the connection to send DMX data to
+     */
+    protected abstract void closeConnection();
+
+    /**
+     * send the buffer of the current universe
+     */
+    protected abstract void sendDmxData();
+
+    /**
+     * install the sending and updating scheduler
+     */
+    protected void installScheduler() {
+        if (senderJob != null) {
+            uninstallScheduler();
+        }
+        if (refreshTime > 0) {
+            senderJob = scheduler.scheduleAtFixedRate(packetSender, 1, refreshTime, TimeUnit.MILLISECONDS);
+            logger.trace("started scheduler for thing {}", this.thing.getUID());
+        } else {
+            logger.info("refresh disabled for thing {}", this.thing.getUID());
+        }
+    }
+
+    /**
+     * uninstall the sending and updating scheduler
+     */
+    protected void uninstallScheduler() {
+        if (senderJob != null) {
+            if (!senderJob.isCancelled()) {
+                senderJob.cancel(true);
+            }
+            senderJob = null;
+            closeConnection();
```

Comment:
> All other calls to closeConnection give other reasons/details.

Hm, [are you sure](https://github.com/eclipse/smarthome/pull/2704/files#diff-4e77f18025e0b189de822236546724adR110)?

> closeConnection checks if the paramater is ThingStatusDetail.NONE and only updates the status if it is not. 

This is quite an unexpected sideeffect of that method and a bad trap for any other contributor that might want to touch the code in future (how shall he know about that behavior and the reason behind it?). Wouldn't it be better to make that explicit, e.g. by having two methods, one that only deals with closing the connection and another that updates the status (and is only called where appropriate)?